lib/ExpensiMark.d.ts

```diff
@@ -0,0 +1,118 @@
+declare type Replacement = string | ((...args: string[]) => string);
+declare type Rule = {
+    name: string;
+    process?: (textToProcess: string, replacement: Replacement) => string;
+    regex?: RegExp;
+    replacement: Replacement;
+    pre?: (input: string) => string;
+    post?: (input: string) => string;
+};
+export default class ExpensiMark {
+    rules: Rule[];
+    htmlToMarkdownRules: Rule[];
+    htmlToTextRules: Rule[];
+    constructor();
+    /**
+     * Replaces markdown with html elements
+     *
+     * @param {String} text - Text to parse as markdown
```

Comment:
@blazejkustra do we need to include types in JSDoc?

Reply:
The convention seems not to include types in JSDoc comments in d.ts files. Below is a code from `d.ts` file in `lodash` from DefinitelyTyped project

```
    interface LoDashStatic {
        /**
         * Creates an array of elements split into groups the length of size. If collection can’t be split evenly, the
         * final chunk will be the remaining elements.
         *
         * @param array The array to process.
         * @param size The length of each chunk.
         * @return Returns the new array containing chunks.
         */
        chunk<T>(array: List<T> | null | undefined, size?: number): T[][];
    }
    ```

Reply:
I removed typings from JSDoc in `.d.ts` files